src/lib.rs

```diff
@@ -143,35 +197,33 @@ impl Mcp2210 {
     }
 }
 
-pub fn scan_devices() -> io::Result<Vec<PathBuf>> {
-    scan_devices_with_filter(|d| {
-        let vendor_id = d.property_value("ID_VENDOR_ID");
-        let model_id = d.property_value("ID_MODEL_ID");
-        vendor_id == Some(OsStr::new("04d8")) && model_id == Some(OsStr::new("00de"))
-    })
+/// Scans devices for the default vendor ID and product ID that the MCP2210 comes with
```

Comment:
Same philosophy here: instead of duplicating the different ways of enumerating/scanning all devices in our API, I'd much rather expose the bare minimum that doesn't push all `hidapi` surprises to our API:

- Expose constant `VENDOR_ID` + `PRODUCT_ID` values if the user wants to use them in a custom filter mechanism
- `pub fn is_mcp2210(info: &DeviceInfo) -> bool` for convenience
- *maybe* also `pub fn open_first(hid: &HidApi) -> Result<Option<Mcp2210>, Mcp2210Error>` for the handling the common case. Note how context creation is still not in our code so we avoid the extra complexity and let the user manage the context in any way they wish (e.g. poll for devices and/or call `refresh_devices` periodically, etc...)

Reply:
I'm not sure I'm clear on what you're wanting. Please 

* Moved the `hidapi_context` creation and management outside this library passing in a reference to it.
* Created `pub const FACTORY_VID` and `pub const FACTORY_PID`. Names ok?
* Created `pub fn is_mcp2210(device_info: &DeviceInfo) -> bool`
* Removed `scan_devices` function.
* Removed `scan_devices_with_filter` function.
* Removed `pub fn scan_devices_with_filter()`.

Reply:
Looks great! :+1: 

That return type of `open_first` is also ok. I just wasn't initially sure what would be the expected result if no such device is connected, but I guess it's now `Mcp2210Error::Hid(HidError::SomeVariant)` (it's not clear from `hidapi` docs what precisely is returned, but that's not a problem)